libtenzir/builtins/operators/serve.cpp

```diff
@@ -663,6 +666,14 @@ struct serve_handler_state {
       TENZIR_ASSERT_CHEAP(ok);
     }
     out_iter = fmt::format_to(out_iter, "}}]}}{}", '\n');
+    if (result.size() > defaults::api::max_response_size) {
+      VAST_WARN("serve-manager discards {} results because the response is too "
+                "big",
+                num_events);
+      return fmt::format(
+        R"({{"next_continuation_token":"{}","events":[], "schemas": []}})",
```

Comment:
We should consider adding a 'warnings' field directly to the response here so this is not happening silently from the FE perspective, but it needs some coordination on how this should be displayed.

Reply:
There is also the pipeline diagnostics format. We should check whether this issue fits into that scheme.